include/power/power_state.h

```diff
@@ -103,6 +103,151 @@ enum pm_state {
 	PM_STATE_SOFT_OFF
 };
 
+/**
+ * Information about a power management state
+ */
+struct pm_state_info {
+	enum pm_state state;
+	int32_t min_residency;
```

Comment:
Make this unsigned.  We're never going to have a valid negative residency, right?  And what does a zero value mean?

Reply:
-1 is option not set, 0 I don't have a concrete use case, but *may* be possible that in some platform the residency time for a specific state is not relevant than 0 can represent it.